### ISSUE TITLE
List pattern is not correctly formatted in the first case of a switch expression

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn#72196]
**Version Used**: '4.10.0-1.24081.13 (fa72fa61)' - Visual Studio 17.10.0 Preview 1.0

Language version: 12.0

**Steps to Reproduce**:

Format the following document in Visual Studio:

```c#
_ = Array.Empty<string>() switch
{
    [] => 0,
    _ => 1,
};
````

The list pattern is push to the left:

````c#
_ = Array.Empty<string>() switch
{
[] => 0,
    _ => 1,
};
````

It seems specific to the list pattern. Other patterns are correctly aligned.

**Expected Behavior**:

The list pattern stays at the same location.

**Actual Behavior**:

The list pattern is pushed to the left

**Comment by SamuelBittmann** [comment]
We have the same issue since .NET SDK 8.0.200. `IDE0055` is reported in this case. One additional observation that I have made: No issue is reported when the list of patterns does not start with a list expression, e.g.:

```csharp
// For this code no warning is reported
var result2 = new int[] { 1, 2 } switch
{
    { Length: 0 } => "empty",
    [1] => "one",
    _ => "unknown"
};
```

Full repro: https://github.com/SamuelBittmann/ide0055_bug_repro
